_plugins/duckdb_functions.rb

```diff
@@ -0,0 +1,149 @@
+require 'json'
+require 'jekyll'
+
+def code(child)
+  "`#{child}`"
+end
+
+def bold(i)
+  "*#{i}*"
+end
+
+def _render_function(function)
+  params = function['parameters']
+  params = params.split(',').map { |it| bold(code(it)) }.join('`, `') if params
+
+  "#{code(function['name'] + '(')}#{params}#{code(')')}"
+end
+
+class Html
+  attr_accessor :html
+
+  def initialize
+    @html = ""
+  end
+
+  def method_missing(m, *args, &block)
+    @html += "<#{m}>"
+    if block_given?
+      instance_eval &block
+    elsif args.length > 0
+      @html += args.join(' ')
+    end
+    @html += "</#{m}>"
+  end
+end
+
+# @return [Array<Object>, nil]
+def get_functions
+  files = Dir.glob('duckdb/src/core_functions/**/*.json')
```

Comment:
globbing is most likely non deterministic, unsure if already implicit but or it would be better to add a sort somewhere.

Reply:
Also I am not actually sure to have sense to have this automatically generated on the fly during deployment vs checked in. I don't see how documentation versioning would work for example.

I would personally probably go either for checking-in the repo a json containing all information (and the pages can then build from that on the fly) OR having directly a script (ruby or anything else) generate the markdown to be checked in.